azurelinuxagent/ga/update.py

```diff
@@ -1494,18 +1488,20 @@ def _reset_legacy_blacklisted_agents(self):
 
 
 class GuestAgent(object):
-    def __init__(self, path=None, pkg=None, is_fast_track_goal_state=False, host=None):
+    def __init__(self, path, pkg, protocol, is_fast_track_goal_state):
```

Comment:
The interface for __init__ is confusing, since it can be used for 2 mutually exclusive purposes depending on the parameters given/omitted. I defined 'from_installed_agent' and 'from_installed_agent' that can be used instead of __init__ with a (hopefully) clearer interface

Reply:
Thanks for changing this